sdk/storage/azure-storage-queue/azure/storage/queue/_shared/authentication.py

```diff
@@ -56,7 +56,7 @@ def _storage_header_sort(input_headers: List[Tuple[str, str]]) -> List[Tuple[str
     sorted_headers = []
     for key in header_keys:
         sorted_headers.append((key, header_dict.get(key)))
-    return sorted_headers
+    return sorted_headers  # type: ignore [return-value]
```

Comment:
`azure\storage\queue\_shared\authentication.py:59: error: Incompatible return value type (got "List[Tuple[str, Optional[str]]]", expected "List[Tuple[str, str]]")  [return-value]`

The `Optional[str]` is from `header_dict.get(key)` and so the only way to coerce the corrct type is `None-check`. So I can either do that or leave this ignore

Reply:
Let's change this to the dictionary method that throws if it doesn't exist
i.e. header_dict[key]

Reply:
or `pop()`